### PR TITLE
Rumor price game mode (frog skin)

### PR DESCRIPTION
**Pull request by klosyandr**

### CODE REVIEW

File: Cauldron/Assets/Scripts/SO/MainSettings.cs [hunk 1]
```diff
@@ -57,6 +58,10 @@ public class StatusBars
             public int ThresholdDecrement => (int)(thresholdDecrement/100 * Total);
             public int GetMinThreshold => (int)(minThreshold/100 * Total);
             public int GetMaxThreshold => (int)((100f - minThreshold)/100 * Total);
+        
+            
+            public event Func<int,int> TryGetCovenCost;
```
Comment (MoreenTissedy) [inline Cauldron/Assets/Scripts/SO/MainSettings.cs]: Well, this looks strange to me)) We actually need a simple int here for override, you can use int? to check it for null.

Using Func looks like over-engineering in our case and I suppose you don't need an event here (or is it used to support several overrides?)

But I'm okey with it)

File: Cauldron/Assets/Scripts/GameDataHandler.cs [hunk 2]
```diff
@@ -91,7 +91,7 @@ public bool IsEnoughMoneyForRumours()
 
         public void BuyRumour()
         {
-            money -= statusSettings.CovenCost;
+            Money -= statusSettings.CovenCost;
```
Comment (MoreenTissedy) [inline Cauldron/Assets/Scripts/GameDataHandler.cs]: Let's fire an event here manually and get on with it)

File: Cauldron/Assets/Scripts/GameDataHandler.cs [hunk 3]
```diff
@@ -135,8 +135,8 @@ private int Set(Statustype type, int newValue)
             int num = newValue - statValue;
             if (num == 0)
                 return statValue;
-            if (type == Statustype.Money && num < 0)
-                return statValue;
+            //if (type == Statustype.Money && num < 0)
```
Comment (MoreenTissedy) [inline Cauldron/Assets/Scripts/GameDataHandler.cs]: I need this><

File: Cauldron/Assets/Scripts/SO/MainSettings.cs (hunk 1 above)
Comment (klosyandr) [reply Cauldron/Assets/Scripts/SO/MainSettings.cs]: I think I used int? here. Not sure if we need another override ("You aren’t gonna need it" principle, yes?).
And... We can't make a struct for settings. "The initializers of the struct fields are not available" to us in the current version of the language - that's what the compiler said...

**Pull request merged**